specs/menu/menu.md

```diff
@@ -0,0 +1,210 @@
+    <fast-menu-item>Menu item two</fast-menu-item>
+    <fast-menu-item>
+        Menu item three
+        <fast-menu slot="item">
+            <fast-menu-item>Nested item one</fast-menu-item>
+            <fast-menu-item>Nested item two</fast-menu-item>
+            <fast-menu-item>
+                Nested item three
+                <fast-menu slot="item">
+                    <fast-menu-item>Nested item one</fast-menu-item>
+                    <fast-menu-item>Nested item two</fast-menu-item>
+                </fast-menu>
+            </fast-menu-item>
+        </fast-menu>
+    </fast-menu-item>
+    <fast-menu-item>Menu item four</fast-menu-item>
+</fast-context-menu>
+```
+
+#### Icon support:
+
+Consider a scenario where we have the following menu - some menu items have glyphs, others do not:
+```html
+<fast-menu>
+    <fast-menu-item>Menu item one</fast-menu-item>
+    <fast-menu-item>Menu item two</fast-menu-item>
+    <fast-menu-item>
+        <fancy-icon slot="icon"></fany-icon>
+        Menu item three
+    </fast-menu-item>
+</fast-context-menu>
+```
+The third item in the menu has a glyph, which would inset the text from the other menu items. To ensure that the content of this menu item aligns with the rest of the items, all items need to visually shift to ensure alignmnet of the menu content. This presents an interesting challenge for how we will communicate these changes to ensure the styling is appropriately applied. Ultimately, the parent just needs to know that *one* of the items below has a menu item with the glyph.
+
+### Prior Art/Examples
+- [FAST-DNA (React)](https://explore.fast.design/components/context-menu)
+- [Material UI](https://material-ui.com/components/menus/)
+- [Lightning Design](https://www.lightningdesignsystem.com/components/menus/)
+- [Ant Design - menu](https://ant.design/components/menu/)
+- [Ant Design - dropdown](https://ant.design/components/menu/)
+- [Atlassian - menu](https://atlaskit.atlassian.com/packages/core/menu)
+- [Atlassian - dropdown menu](https://atlaskit.atlassian.com/packages/core/dropdown-menu)
+- [Windows (UWP)](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/tree-view)
+
+---
+
+## Design
+
+### API
+
+*The key elements of the component's public API surface:*
+
+**Menu**
+*Component name:*
+- `fast-menu`
+
+*Attributes:*
+- extends HTML Element attributes
+
+*Slots:*
+- default slot for items
+
+*Events*
+- none
+
+**Menu Item**
+*Component name:*
+- `fast-menu-item`
+
+*Attributes:*
+- role - an enum representing the menu items' role
+    - menuitem
+    - menuitemcheckbox
+    - menuitemradio
+- disabled - the menu item is disabled
+- checked - sets the checked value for menuitemcheckbox or menuitemradio items
+
+*Slots:*
+- content - slot for the content (the default slot for the item)
+- item - the optional children
+- before-content - slot which precedes content, often a glyph or icon
+- after-content - slot which follows content, often a glyph or icon
```

Comment:
I know we had some of this for tree-view-item as well...but to remind me: was this because we needed to align the before, content, and after in a columnar layout? If not, it seems we could just let folks put whatever they want in content.

Reply:
We synced on this offline - I'm going to look and see if there is an abstract utility to support simply passing content and having a class create this kind of grid.

Reply:
I've removed references to this here for now. If we can come up with an abstract css utility to manage these I think we can do without. If not, the slots will reappear.

Reply:
@EisenbergEffect @nicholasrice @janechu I think we'll need these slots. I'd like to have a quick sync sometime today to discuss my very abbreviated and short/initial test with not having these named slots.